docs/src/design/stack/io_ops.md

```diff
@@ -0,0 +1,193 @@
+
+$$
+v_{old} = \sum_{i=0}^3\alpha_{i+4} \cdot s_{3-i}'
+$$
+
+The value representing state of memory after the operation:
+
+$$
+v_{new} = \sum_{i=0}^3\alpha_{i+8} \cdot s_{3-i}'
+$$
+
+Note that since this is a _read_ operation, the values come from the same registers, but the $\alpha$ values are different from the values used to compute the old state of memory.
+
+Using the above variables, we define the value representing the memory access request as follows:
+
+$$
+u_{mem} = \alpha_0 + \alpha_1 \cdot op_{mem} + \alpha_2 \cdot s_0 + \alpha_3 \cdot clk + v_{old} + v_{new}
+$$
+
+In the above:
+- $op_{mem}$ is the unique [operation label](../chiplets/main.md#operation-labels) of the memory access operation.
+- $s_0$ is the memory address from which the values are to be loaded onto the stack.
+- $clk$ is the current clock cycle of the VM.
+
+The effect of this operation on the rest of the stack is:
+* **Left shift** starting from position $5$.
+
+### MLOAD
+Assume that the words with elements $v_0, v_1, v_2, v_3$ is located in memory at address $a$. The `MLOAD` operation pops an element off the stack, interprets it as a memory address, and pushes the first element of the word located at the specified address to the stack. The diagram below illustrates this graphically.
+
+![mload](../../assets/design/stack/io_ops/MLOAD.png)
+
+To simplify description of memory access request value, we first define the following variables:
+
+The value representing state of memory before the operation (values in registers $h_0, h_1, h_2$ are set by the prover non-deterministically):
+
+$$
+v_{old} = \alpha_4 \cdot s_0' + \sum_{i=2}^3\alpha_{i+4} \cdot h_i'
+$$
+
+The value representing state of memory after the operation:
+
+$$
+v_{new} = \alpha_8 \cdot s_0' + \sum_{i=0}^2\alpha_{i+8} \cdot h_i'
+$$
+
+Using the above variables, we define the value representing the memory access request as follows:
+
+$$
+u_{mem} = \alpha_0 + \alpha_1 \cdot op_{mem} + \alpha_2 \cdot s_0 + \alpha_3 \cdot clk + v_{old} + v_{new}
+$$
+
+In the above:
+- $op_{mem}$ is the unique [operation label](../chiplets/main.md#operation-labels) of the memory access operation.
+- $s_0$ is the memory address from which the value is to be loaded onto the stack.
+- $clk$ is the current clock cycle of the VM.
+
+The effect of this operation on the rest of the stack is:
+* **No change** starting from position $1$.
+
+### MSTOREW
+Assume that the words with elements $v_0, v_1, v_2, v_3$ is located in memory at address $a$. The `MSTOREW` operation pops an element off the stack, interprets it as a memory address, and writes the remaining $4$ elements at the top of the stack into memory at the specified address. The stored elements are not removed from the stack. The diagram below illustrates this graphically.
+
+![mstorew](../../assets/design/stack/io_ops/MSTOREW.png)
+
+After the operation the contents of memory at address $a$ would be set to $u_0, u_1, u_2, u_3$.
```

Comment:
nit: This is backwards from how we specify old/new values in the Memory chiplet design docs, which I think is a bit confusing. (Over there we use `u` for old and `v` for new.) I'd prefer to be consistent with that or to change that to be consistent with this.

Reply:
Agreed! I've updated them here to match how it is in memory chiplet description.